### ISSUE TITLE
Build with Strider instead of Travis

### ISSUE THREAD

**Issue by smashwilson** [deconst/deconst-docs#1]
@rgbkrk: Let's use this repo for administrative and architecture discussions, too. It makes a certain amount of sense, because we should update the docs when we update the architecture, once we have docs, and once we have an architecture :grinning: 

On that note: I've been thinking that we'll basically need to use something self-hosted for CD. The reason:
- We need to have a PR builder that publishes a preview to Cloud Files and posts a link back to the PR.
- The PR builder needs some kind of credentials to publish the preview.
- If we use Travis, anyone can trivially change the `.travis.yml` to run `env` or something on their branch, and yank whatever API key we drop in there.

In other words, Travis' secrets protection works by only exposing secrets on trusted branches like `master`, but we need secrets on pull requests, too. Self-hosting CD will let us lock the project build to _only_ run `deconst-render-rst`.

I'd personally prefer to go with [Strider](https://github.com/Strider-CD/strider) rather than Jenkins because Jenkins is kind of heavy and awkward. I've been playing around with it a bit today and it certainly meets our base requirement for "accept a GitHub webhook and run some bash".

**Comment by smashwilson** [comment]
As per @rgbkrk: I'm shelving this for the moment so that we don't get hung up on infrastructure while we're trying to ship, especially because the PR builder isn't in 0.1.

**Comment by rgbkrk** [comment]
I think the build out with strider is making things more complex way too early. I personally think we should rely on things like Travis and use webhooks _when_ we need those extra builds for PRs, branches, etc.

I realize how Travis' secrets protection works, I spoke on it at DEF CON. I actually feel comfortable in using it since it's:
- auditable (assuming no problems in Travis infra)
- fairly easy to use
- already deployed and ready for us

Since strider accepts webhooks, if we want more complex things to happen after, well...

**Comment by rgbkrk** [comment]
@smashwilson Cool, thanks. I didn't want to come in starting a :fire:, it just seemed a bit much at the start to digest.

**Comment by smashwilson** [comment]
[Drone](https://github.com/drone/drone) is also worth a look as a contender.

Basically, my criteria for this are:
- It should be able to accept a webhook and run some kind of script. (Basically any CI system will do this because that's what CI _is_.)
- It should work relatively well with automation - I'd like to use Ansible to install it and create users. This is where Jenkins is awkward, I feel.
- It should have an API we can use to create jobs automatically from other system components, so we can do the automated job setup described in #8. Travis is awkward here: we would need to automatically create pull requests to add `.travis.yml` files, which is certainly possible, but feels strange.
- Bonus points for OAuth integration so we can log in with GitHub credentials rather than remembering yet another password.

**Comment by smashwilson** [comment]
Note to self: maintain the ability to run other linters and checkers during the "build" process.
